tests/ConfigTest.php

```diff
@@ -85,4 +85,16 @@ public static function returnsDefaultProvider(): array
             ],
         ];
     }
+
+    public function testIfCachePersistsAfterContentChange(): void
```

Comment:
Caching should be removed, so this test can go too.

It also wasn't really a proper test IMO. Since this PR makes `Config` state full, the entire test suite should pass with both a cache enabled config, and when it is disabled.